src/main/java/com/google/cloud/teleport/templates/common/SplunkConverters.java

```diff
@@ -220,11 +221,17 @@ public void processElement(ProcessContext context) {
                               builder.withIndex(index);
                             }
 
+                            String event = metadata.optString(HEC_EVENT_KEY);
+                            if (!event.isEmpty()) {
+                              builder.withEvent(event);
+                            }
                             // We remove the _metadata entry from the payload
                             // to avoid duplicates in Splunk. The relevant entries
                             // have been parsed and populated in the SplunkEvent metadata.
                             json.remove(METADATA_KEY);
-                            builder.withEvent(json.toString());
+                            if (event.isEmpty()) {
```

Comment:
Add a comment here so it's easier for someone who is not familiar with the _metadata overrides to understand why we are doing this. 

Otherwise this looks good!

Reply:
Done